bluesky/run_engine.py

```diff
@@ -1730,3 +1724,15 @@ def _default_md_validator(md):
             "GOOD: sample='dirt' "
             "GOOD: sample={'color': 'red', 'number': 5} "
             "BAD: sample=[1, 2] ")
+
+def _try_stopping(obj):
+    "Call obj.stop() and log any exceptions."
+    try:
+        stop = obj.stop
+    except AttributeError:
+        logger.debug("No 'stop' method available on %r", obj)
+    else:
+        try:
+            stop()
+        except Exception:
+            logger.error("Failed to stop %r", obj)
```

Comment:
That doesn't really answer the question though - shouldn't we know why it failed to stop?